astropy/io/ascii/tests/test_ecsv.py

```diff
@@ -631,94 +629,80 @@ def test_read_not_json_serializable():
         Table.read(txt, format='ascii.ecsv')
 
 
-def test_read_complex():
-    """Test an ECSV file with a complex column"""
+def test_read_bad_datatype():
+    """Test a malformed ECSV file"""
     txt = """\
 # %ECSV 1.0
 # ---
 # datatype:
-# - {name: a, datatype: complex}
+# - {name: a, datatype: object}
 # schema: astropy-2.0
 a
-1+1j
-2+2j"""
-    match = "datatype 'complex' of column 'a' is not in allowed values"
-    with pytest.raises(ValueError, match=match):
-        Table.read(txt, format='ascii.ecsv')
+fail
+[3,4]"""
+    with pytest.warns(InvalidEcsvDatatypeWarning,
+                      match="unexpected datatype 'object' of column 'a' is not in default datatypes"):
+        t = Table.read(txt, format='ascii.ecsv')
+    assert t['a'][0] == "fail"
+    assert type(t['a'][1]) is str
```

Comment:
Yeah, so this is part of the original rationale for being strict, namely that this "object" type doesn't really round trip in the expected way. Anyway, the warning should do the trick.